packages/core/src/lib/components/T/T.ts

```diff
@@ -74,4 +66,8 @@ export const T = new Proxy(TComp, {
 
     return TComp
   }
-}) as typeof TComp & TComponentProxy & Extensions
+}) as typeof TComp & {
+  [Key in keyof typeof THREE]: Component<Props<(typeof THREE)[Key]>, {}, 'ref'>
+} & {
+  [Key in keyof Threlte.UserCatalogue]: Component<Props<Threlte.UserCatalogue[Key]>, {}, 'ref'>
+} & Extensions
```

Comment:
nice!

Reply:
This is trying to fix some core type issues, but I'm going to leave this for a later PR.

Reply:
Fixed here: https://github.com/threlte/threlte/pull/1580